### DISCUSSION TITLE
Serial performance of USB devices?

### DISCUSSION THREAD

**Discussion by andyp1per** [dorssel/usbipd-win#306] [Q&A] [answered]
I have it all working and have been able to connect to an STM32 based flight controller using usbipd on Windows 10 / WSL2. So first of all great job!

My question is how to make it go faster? The serial connection is very slow - much slower than a direct connection on windows using the same software and I am wondering if there are any settings I can change (e.g. via stty) to speed things up. I am connecting to /dev/ttyACM0

Thx - Andy

**Comment by dorssel** [comment]
You may be suffering from issue #300. If you can, please follow the instructions in that thread to capture `usbmon` debugging traces. It may help solve the issue.

**Reply by andyp1per** [reply]
Hi here is the pcap file
[kakute.zip](https://github.com/dorssel/usbipd-win/files/8257521/kakute.zip)

**Reply by dorssel** [reply]
Yep, this makes sense. The capture shows the same problematic behavior as #300. Glad it is fixed now. Thanks for the report!

**Comment by andyp1per** [comment, marked as answer]
Hey @dorssel your fix https://github.com/dorssel/usbipd-win/pull/307 cures this for me! Thanks!
